### ISSUE TITLE
Enable task for Renewal Email Notifications

### ISSUE THREAD

**Issue by moorerst** [cityofaustin/atd-data-tech#4844]
Review triggers and email content for Renewal Notifications
Determine if task should be enabled now
Questions 
- since we are at the back end of this current renewal period should we wait for the next renewal period to enable this task?
- what are the criteria for this task to generate a email notification? My concern is I don't want residents who have already renewed to receive a notification.

**Comment by johnclary** [comment]
These are triggered by permit expiration date. Next renewal period starts in August. Next notifications needs to out in July

**Comment by dianamartin** [comment]
Been working with Stephanie so that she can help review the email content and the email tasks. Have had a few iterations already, feeling very close.

**Comment by dianamartin** [comment]
The `Renewal Year` (January/August) field exists on the `permit` object table
- Created new field `Renewal Period` on `permit_application` object table to bring in that field from the `permit` table

**Comment by dianamartin** [comment]
Talked to Chrispin about `equation` operators, will do some testing to see if I can get the Renewal Due Date to auto-populate based on existing fields in RPP.

**Comment by dianamartin** [comment]
@moorerst @ChrispinP 

Did some testing and got the result of a renewal deadline that I wanted, however, it occurred to me that an individual is going to have multiple applications records and a set number of permits that were issued. Would we be updating the previous issued permits or issuing out 4 new permits based on the renewal? (see example table below)

Ex.

| Created | App | Permits | Issued | Expiration |
| --- | ---  | --- | --- | --- |
| 8/13/20| New - App 1 | **Permits tied to this application** | - |
| - | - | Permit 1 | 8/13/20 | 8/13/21 |
| - | - | Permit 2 | 8/13/20 | 8/13/21 |
| - | - | Permit 3 | 8/13/20 | 8/13/21 |
| - | - | Permit 4 | 8/13/20 | 8/13/21 |
| 12/29/20 | Renewal - App 2 | **No permits tied to this application** | - |

**Comment by moorerst** [comment]
@dianamartin I think we would be updating the new permits based on what is requested in the Renewal application.

**Comment by ChrispinP** [comment]
@dianamartin the equation would update its value for all records. we therefore need it to apply fairly no matter the date or year the permit was issued. We may want to add a boolean field "Is Expired" or just "Expired" that checks that expiration date and updates accordingly. If all permits for an application are expired, it would not send a renewal notification. It may also be helpful to create a day countdown - a date equation set to days that calculates the difference between expiration date and current date. once day countdown = 31 then it sends out that renewal email.

as Steph mentions, a renewal application may have the same number of permits as the previous year but they could request less or more on their application for the new year.

additionally, its possible that a customer may have multiple active applications due to buying additional or replacement permits. we would want to check if these two applications have different expirations, and if not, how we can combine the renewal notification for multiple applications with the same expiration.

**Comment by dianamartin** [comment]
@moorerst @ChrispinP I think that the renewal process needs to be fully developed before we start making assumptions on what we are going to do with the renewal email. It's one small portion of this renewal process. 

Where has the conversation for the RPP renewals gone? Should we discuss at the upcoming RPP meeting we have next week?

**Comment by moorerst** [comment]
@dianamartin let me know when you have some time to discuss. Thanks

**Comment by dianamartin** [comment]
@moorerst Thursday is more open than Wed this week :)

**Comment by moorerst** [comment]
@dianamartin Question to RPP staff regarding renewals that happen outside of the renewal period.
If the permit is schedule to be renewed in Aug but it is not renewed until Nov what would be the new Renewal Period for that permit?
Per Steve - Because we don’t do any pro-rating, the permits would still only be good until August ( July 31st )

**Comment by moorerst** [comment]
discuss at next sprint review

**Comment by moorerst** [comment]
Sprint Review scheduled for 3/11

**Comment by dianamartin** [comment]
# Sprint Review Notes about Renewals
## Jan
Calculate whichever is easier, 60 days or a set date
- 60 days before/ Nov 1 
- 30 days before/ Dec 1
Due Date: Dec 31

## August
Calculate whichever is easier, 60 days or a set date
- 60 days before/ Jun 1
- 30 days before/ Jul 1
Due Date: July 31

Steve: to share expiration email template we made a while back

Had concerns about users that don't have email addresses in the system

**Page or filter** 
- Find people that don't use email
- People didn't notify them at all
- In the past they weren't contacted, responsibility of resident to get a permit

Renewal questions from residents: Information being saved - re-entering everything?

**Comment by dianamartin** [comment]
@dianamartin Schedule time to do this!

**Comment by dianamartin** [comment]
# User Application Example
User applied 3/31/2021
Transaction in 4/2021
Added permit 4/9/2021
JANUARY RENWAL
Due Date 12/31/21
30 day 12/1/21
60 day 11/1/21

User applied 5/31/2021
Transaction in 6/2021
Added permit 6/9/2021
AUGUST RENWAL
Due Date 8/31/22
30 day 7/1/22
60 day 6/1/22

**Comment by dianamartin** [comment]
I had been testing in the TEST RPP environment, adding some fields to see if I can get the due dates correctly. Still work in progress.

**Comment by ChrispinP** [comment]
Created the May 14 Test env to help test Renewal Emails

**Comment by ChrispinP** [comment]
My Test Build:
![image](https://user-images.githubusercontent.com/67915167/118729319-48bdba80-b7fb-11eb-9df4-ce8a708cdbe0.png)

1) Renewal Date is difficult to set automatically, ideally, this should be set manually by updating after Transaction Date is established based on what Renewal Period the customer applies for (Jan vs Aug) and whether they are applying for current period or the next 12 month period.

2) 2 Equations determine the First and Second Email Dates from the Renewal Data for data purposes. These are set to Dates.
![image](https://user-images.githubusercontent.com/67915167/118729816-3001d480-b7fc-11eb-9f93-2c3013e24bf1.png)

3) Days Until Renewal Date is the field used to set the Email Task when at 60 and 30 days before Renewal Date
![image](https://user-images.githubusercontent.com/67915167/118729691-f8932800-b7fb-11eb-9215-f68671f43486.png)

4) Count fields count the number of application records connected to the customer where Transaction Date is within a certain block of months. These can be used as points of reference for staff to see how many applications the customer submitted during a certain period.
![image](https://user-images.githubusercontent.com/67915167/118729445-8d495600-b7fb-11eb-8033-2a8170395696.png)

5) Staff can also use the Renewal Period pulled from the zone to the application as a reference point as well when updating the Renewal Date on the Customer Record

6) We will also need an expiration task that removes the Renewal Date value if Jan 1st or Aug 1st passes and the Customer has not submitted an application within the last 2 years/ 24 months so they are no longer on the email blast.

**Comment by moorerst** [comment]
manual email blast as an option
download list of customer to be notified - new issue
